src/Sulu/Bundle/CategoryBundle/Infrastructure/Sulu/Search/CategoryReindexProvider.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Sulu.
+ *
+ * (c) Sulu GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Bundle\CategoryBundle\Infrastructure\Sulu\Search;
+
+use CmsIg\Seal\Reindex\ReindexConfig;
+use Doctrine\ORM\EntityManagerInterface;
+use Doctrine\ORM\EntityRepository;
+use Sulu\Bundle\CategoryBundle\Entity\CategoryInterface;
+use Sulu\Bundle\CategoryBundle\Entity\CategoryTranslationInterface;
+
+/**
+ * @phpstan-type Category array{
+ *     id: int,
+ *     changed: \DateTimeImmutable,
+ *     created: \DateTimeImmutable,
+ *     translation: string,
+ *     locale: string,
+ * }
+ */
+class CategoryReindexProvider
```

Comment:
Make `@internal` and final see comments in https://github.com/sulu/sulu/pull/8227